File: 09_factoresPrimos/Java/descFacPrim.java
import java.util.Scanner;
public class descFacPrim
{
    public static boolean primo(int k)
    {
        int i;
        for(i=2;i<k;i++)
        {
            if(k%i==0)
                return false;
        }
        return true;
    }
    public static int sigp(int k)
    {
        k++;
        while(!primo(k))
            k++;
        return k;
    }
    public static void main(String[] args)
    {
        Scanner leer=new Scanner(System.in);
        int n,j;
        System.out.print("Ingresa un numero: ");
        n=leer.nextInt();
        if(primo(n))
            System.out.println("Es primo\n");
        else
        {
            j=2;
            do
            {
                while(n%j!=0)
                    j=sigp(j);
                System.out.print("("+j+")");
                n=n/j;
            }while(!primo(n));
            System.out.println("("+n+")");
        }
    }
}

File: 09_factoresPrimos/Python/descFacPrim.py
def primo(k):
    for i in range(2,k):
        if k%i==0:
            return False
    return True
def sigp(h):
    h=int(h+1)
    while (not primo(h)):
        h=int(h+1)
    return h
n=int(input("Ingresa un numero: "))
if primo(n):
        print("Es primo\n")
else:
    j=int(2)
    while True:
        while n%j!=0:
            j=sigp(j)
        print("(%d)"%(j))
        n=int(n/j)
        if primo(n):
            break
    print("(%d)"%(n))